src/compiler/checker.ts

```diff
@@ -5747,6 +5747,10 @@ namespace ts {
                 markAliasSymbolAsReferenced(symbol);
             }
 
+            if (symbol.flags & SymbolFlags.Enum) {
+                checkEnumIdentifier(node, symbol);
+            }

```

Comment:
The previous placement (checkExpression) was correct. If you put it here, you will not catch dotted names.
```ts
namespace N {
    export const enum E { }
    var v;
}
N.E; // Should be error
```